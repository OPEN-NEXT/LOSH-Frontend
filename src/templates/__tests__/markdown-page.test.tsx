import * as React from "react";
import { useStaticQuery } from "gatsby";
import ShallowRenderer from "react-test-renderer/shallow";
import Template from "../markdown-page";
import { markdownQuery } from "../__mocks__/markdown-page-query.mock";

const renderer = ShallowRenderer.createRenderer();

jest.mock("antd", () => {
	return {
		Typography: {
			Title: (): null => null,
		},
	};
});

jest.mock("../../components/layout", () => {
	const Layout = (): null => null;
	return Layout;
});

jest.mock("../../components/seo", () => {
	const SEO = (): null => null;
	return SEO;
});

describe("Markdown Page", () => {
	it("renders correctly", () => {
		const tree = renderer.render(
			<Template data={markdownQuery.data}></Template>
		);
		expect(tree).toMatchSnapshot();
	});
});
