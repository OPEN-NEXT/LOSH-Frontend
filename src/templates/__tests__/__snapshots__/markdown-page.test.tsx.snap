// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Page renders correctly 1`] = `
<Layout>
  <SEO
    title="About our project"
  />
  <Title
    level={1}
  >
    About our project
  </Title>
  <div
    className="markdown-content"
  >
    <div>
      <p>
        The Library of Open Source Hardware was created by the 
        <a
          href="https://opennext.eu/"
        >
          OPEN!NEXT
        </a>
         working group. They are a team of institutions and organizations that support the growth of Open Source hardware. Tasked by the 
        <a
          href="https://cordis.europa.eu/project/id/869984"
        >
          European Commission
        </a>
         to create a consolidated database for Open Source Hardware specifications, the OPEN!NEXT working group created this library for use by the maker community.
      </p>
      

      <p>
        These organizations are part of the working group:
      </p>
      

      <ul>
        

        <li>
          <a
            href="https://wikimedia.de"
          >
            Wikimedia Germany
          </a>
        </li>
        

        <li>
          <a
            href=""
          >
            Fraunhofer IPK
          </a>
        </li>
        

        <li>
          <a
            href=""
          >
            TU Berlin
          </a>
        </li>
        

        <li>
          <a
            href=""
          >
            Wikifactory
          </a>
        </li>
        

      </ul>
      

      <h3>
        Our mission
      </h3>
      

      <ol>
        

        <li>
          Development of standards for the open source community to enable sharing and reuse of data and hardware
        </li>
        

        <li>
          Providing an accessible open source database that allows for easy exploration of the stored data and guides makers in contributing their open source hardware specifications.
        </li>
        

      </ol>
    </div>
  </div>
</Layout>
`;
